### PR TITLE
defaultsDeep handle undefined

### PR DESCRIPTION
**Pull request by FredericHeem**

### CODE REVIEW

**Comment by richytong** [comment]
Hello, I believe the value == null check also accounts for undefined

**Comment by richytong** [comment]
Also looks like optional chaining isn't supported in node 12

https://stackoverflow.com/questions/59574047/how-to-use-optional-chaining-in-node-js-12

File: x/defaultsDeep.js [hunk 1]
```diff
@@ -51,7 +51,7 @@ const objectDefaultsDeepFromObject = function (object, defaultObject) {
       result[key] = arrayDefaultsDeepFromArray(item, defaultItem)
     } else if (item == null) {
       result[key] = defaultItem
-    } else if (item.constructor == Object && defaultItem.constructor == Object) {
+    } else if (item.constructor == Object && defaultItem && defaultItem.constructor == Object) {
```
Comment (richytong) [inline x/defaultsDeep.js:54]: could we make the defaultItem part of this check defaultItem != null?

File: x/defaultsDeep.js [hunk 2]
```diff
@@ -107,7 +107,7 @@ const defaultsDeep = defaultCollection => function defaulting(value) {
   if (value == null) {
     return value
   }
-  if (value.constructor == Object && defaultCollection.constructor == Object) {
+  if (value.constructor == Object && defaultCollection && defaultCollection.constructor == Object) {
```
Comment (richytong) [inline x/defaultsDeep.js:110]: same thing with defaultCollection, could we make it defaultCollection != null?

**Review by richytong — CHANGES_REQUESTED** [review]

**Review by richytong — CHANGES_REQUESTED** [review]
It looks like if the default object is null we should just return the argument like the identity function so that check should go in the top level defaultsDeep. Could you change the pr so that it checks for it there?

**Comment by richytong** [comment]
I think we can do the check even before we return the function defaulting. If the default value is null return identity instead of defaulting.

File: x/defaultsDeep.js [hunk 3]
```diff
@@ -107,7 +111,7 @@ const defaultsDeep = defaultCollection => function defaulting(value) {
...
```
[the remark this answers is not in the document]
Comment (FredericHeem) [reply x/defaultsDeep.js]: done

**Review by richytong — APPROVED** [review]
LGTM

**Pull request merged**